server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/BalancedShardsAllocator.java

```diff
@@ -96,17 +97,19 @@ public class BalancedShardsAllocator implements ShardsAllocator {
 
     private volatile WeightFunction weightFunction;
     private volatile float threshold;
+    private final WriteLoadForecaster writeLoadForecaster;
```

Comment:
Just to check my understanding, we need this here so that we can go via `WriteLoadForecaster#getForecastedWriteLoad` to include the license check right?

Reply:
Yes, that's correct.